TheMovieDb/Views/HomeViewController/HomeViewController.swift

```diff
@@ -0,0 +1,195 @@
+          layoutSize: headerSize,
+          elementKind: "HeaderCollectionView.reuseIdentifier",
+          alignment: .top)
+        
+        let section = NSCollectionLayoutSection(group: group)
+        section.boundarySupplementaryItems = [sectionHeader]
+        section.orthogonalScrollingBehavior = .groupPaging
+        
+        return UICollectionViewCompositionalLayout(section: section)
+    }()
+    
+    private var latestSearch: String?
+    lazy private var searchController: SearchBar = {
+        let searchController = SearchBar("Search a Movie", delegate: self)
+        searchController.text = latestSearch
+        searchController.showsCancelButton = !searchController.isSearchBarEmpty
+        return searchController
+    }()
+    
+    var movieHomeCollectionView: UICollectionView!
+   
+    var tableView: UITableView!
+  
+    func showEmptyState() {
+        print("")
+    }
+    
+    var presenter: HomeViewPresenter?
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Customize navigation bar.
+        guard let navbar = self.navigationController?.navigationBar else { return }
+        self.navigationItem.title = "Movies"
+        navbar.tintColor = .black
+        navbar.titleTextAttributes = [.foregroundColor: UIColor.black]
+        navbar.prefersLargeTitles = true
+
+        // Set up the searchController parameters.
+        navigationItem.searchController = searchController
+        definesPresentationContext = true
+        presenter?.fetchAllMovieList()
+    }
+    
+    func setUpCollectionView() {
+        let collectionView = UICollectionView(frame: view.bounds, collectionViewLayout: compositionalLayout)
+        view.addSubview(collectionView)
+        collectionView.autoresizingMask = [.flexibleHeight, .flexibleWidth]
+        collectionView.backgroundColor = .systemBackground
+        collectionView.register(LargeCollectionViewCell.self, forCellWithReuseIdentifier: LargeCollectionViewCell.reuseIdentifer)
+        collectionView.register(
+          HeaderCollectionView.self,
+          forSupplementaryViewOfKind: "HeaderCollectionView.reuseIdentifier",
+          withReuseIdentifier: HeaderCollectionView.reuseIdentifier)
+        let refreshControl = UIRefreshControl()
+        refreshControl.addTarget(self, action: #selector(refresh), for: .valueChanged)
+        collectionView.refreshControl = refreshControl
+        collectionView.sendSubviewToBack(refreshControl)
+        movieHomeCollectionView = collectionView
+    }
+    
+    @objc func refresh() {
+        presenter?.fetchAllMovieList()
+    }
+    
+    func showMoviesHome(arrMovie: [SectionMovie: [Movie]]) {
+        self.arrMovies = arrMovie
+        setUpCollectionView()
+        configureDataSource()
+        movieHomeCollectionView.reloadData()
+    }
+    
+    func showLoading() {
+        print("")
+    }
+    
+    func stopLoading() {
+       
+    }
```

Comment:
If this is meant to be updated in the future please try to add `// TODO:` comments in the method's body.